### PR TITLE
[PM-9631] MP reprompt view items

### PR DESCRIPTION
**Pull request by Jingo88**
## 🎟️ Tracking

[PM-9631](https://bitwarden.atlassian.net/browse/PM-9631)

## 📔 Objective

This ticket has changed in scope. We will no longer include MP reprompt within the view of an item. Instead if an item has MP reprompt turned on we will show it on click in the vault. 

- Have MP reprompt appear on a item in the vault view
- Remove MP reprompt from:
  - View - Attachments Download
  - View - Edit Button
  - View - Delete Button
- Other Areas for MP Reprompt in the vault list page
  - Autofill
  - Fill and save
  - Clone
  - Assign to Collections (Vault page 3 dot menu)
  - Copy Actions - except username

## 📸 Screen Recording

https://github.com/user-attachments/assets/78d00ed4-4450-43ad-8d21-371b0ad73b4b

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-9631]: https://bitwarden.atlassian.net/browse/PM-9631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html [hunk 1]
```diff
@@ -28,12 +28,7 @@
       <a routerLink="" bitMenuItem (click)="clone()">
         {{ "clone" | i18n }}
       </a>
-      <a
-        routerLink="/assign-collections"
-        [queryParams]="{ cipherId: this.cipher.id }"
-        type="button"
-        bitMenuItem
-      >
+      <a routerLink="" type="button" bitMenuItem (click)="assignToCollections()">
```
Comment (nick-livefront) [inline apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html:31]: - ⛏ `type="button"` isn't applicable to an anchor (pre-existing to your changes)
- ❓ Is there a benefit to including an empty `routerLink`?

File: libs/vault/src/services/copy-cipher-field.service.ts [hunk 2]
```diff
@@ -58,10 +58,10 @@ const CopyActions: Record<CopyAction, CopyActionInfo> = {
...
     event: EventType.Cipher_ClientCopiedCardCode,
   },
-  email: { typeI18nKey: "email", protected: false },
-  phone: { typeI18nKey: "phone", protected: false },
-  address: { typeI18nKey: "address", protected: false },
-  secureNote: { typeI18nKey: "note", protected: false },
+  email: { typeI18nKey: "email", protected: true },
+  phone: { typeI18nKey: "phone", protected: true },
+  address: { typeI18nKey: "address", protected: true },
+  secureNote: { typeI18nKey: "note", protected: true },
```
Comment (nick-livefront) [inline libs/vault/src/services/copy-cipher-field.service.ts:64]: This is probably for my own knowledge, how is `protected` used?

**Review by nick-livefront — COMMENTED** [review]
🚀 A single nit for me, otherwise questions!

File: libs/vault/src/services/copy-cipher-field.service.ts (hunk 2 above)
Comment (Jingo88) [reply libs/vault/src/services/copy-cipher-field.service.ts:64]: Initially this was built with the intention that only some fields in the copy list would be `protected` and checked for master password reprompt. Now we're applying the MP reprompt to all fields except `username`. The logic is checked once in the `copy` method if statement

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html (hunk 1 above)
Comment (Jingo88) [reply apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html:31]: ah great callout. i don't think there is a benefit. i'll get rid of them

File: libs/vault/src/services/copy-cipher-field.service.ts (hunk 2 above)
Comment (nick-livefront) [reply libs/vault/src/services/copy-cipher-field.service.ts:64]: Ah, I was looking at the consumers to where it was used. Thanks for the explanation!

**Review by nick-livefront — DISMISSED** [review]

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html [hunk 3]
```diff
@@ -25,7 +25,7 @@
       {{ favoriteText | i18n }}
     </button>
     <ng-container *ngIf="canEdit">
-      <a routerLink="" bitMenuItem (click)="clone()">
+      <a bitMenuItem (click)="clone()">
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html:28]: 👍 Thanks!

File: apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.ts [hunk 4]
```diff
@@ -100,9 +100,22 @@ export class VaultListItemsContainerComponent {
...
     private vaultPopupAutofillService: VaultPopupAutofillService,
+    private passwordRepromptService: PasswordRepromptService,
+    private router: Router,
   ) {}
 
   async doAutofill(cipher: PopupCipherView) {
     await this.vaultPopupAutofillService.doAutofill(cipher);
   }
+
+  async onViewCipher(cipher: any) {
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.ts:111]: ⛏️ Please avoid `any` type when possible. I think we can use `PopupCipherView` in this case.

**Review by shane-melton — CHANGES_REQUESTED** [review]
Looks good! Just one small nit.

File: apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.ts (hunk 4 above)
Comment (Jingo88) [reply apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.ts]: ah i missed that. thanks

**Review by shane-melton — APPROVED** [review]
Looks good, nice work!

**Pull request merged**